torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1011,7 +1070,9 @@ def _maybe_evaluate_static(self, expr: "sympy.Expr") -> "Optional[sympy.Expr]":
         floor_div_replace = {}
         for atom in new_expr.atoms(FloorDiv):
             floor_div_replace[atom] = sympy.floor(atom.args[0] / atom.args[1])
-        new_expr = safe_expand(new_expr.xreplace(floor_div_replace))
+        # NB: we use subs and not xreplace to simplify all FloorDiv sub-exprs,
+        # like: FloorDiv(6.28, (FloorDiv(6.28, 3.14)))
+        new_expr = safe_expand(new_expr.subs(floor_div_replace))
```

Comment:
No regression here as there are no significant changes to `FloorDiv`. Only needed because of the new `test_floordiv_simplify` test, which exposed this issue